core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -701,16 +666,15 @@ private[spark] class BlockManager(
       blockId: BlockId,
       values: Iterator[Any],
       level: StorageLevel,
-      tellMaster: Boolean = true,
-      effectiveStorageLevel: Option[StorageLevel] = None): Boolean = {
+      tellMaster: Boolean = true): Boolean = {
     require(values != null, "Values is null")
-    val result = doPut(
+    val maybeIterator = doPutIterator(
       blockId,
-      IteratorValues(() => values),
+      () => values,
       level,
-      tellMaster,
-      effectiveStorageLevel)
-    result == DoPutSucceeded
+      tellMaster)
+    // If doPut() didn't hand work back to us, then block already existed or was successfully stored
+    maybeIterator.isEmpty
```

Comment:
I just removed the variable, since the entire line and the `isEmpty` fit together.